packaging/Microsoft.NET.Sdk.WindowsDesktop/targets/Microsoft.NET.Sdk.WindowsDesktop.targets

```diff
@@ -27,11 +27,26 @@
 
   <ItemGroup Condition=" ('$(EnableDefaultItems)' == 'true') And ('$(UseWPF)' == 'true') And ('$(_TargetFrameworkVersionValue)' != '$(_UndefinedTargetFrameworkVersion)') And 
                          ('$(_TargetFrameworkVersionValue)' >= '$(_WindowsDesktopSdkTargetFrameworkVersionFloor)')">
-    
-    <!-- In the WindowsDesktop .props, we globbed all .xaml files as Page items.  If any of those files are included
-         as Resource, Content, or None items, then remove them from the Page items. -->
-    <Page Remove="@(Resource);@(Content);@(None)"
+
+    <!--
+         In the WindowsDesktop .props, we globbed all .xaml files as Page items.  If any of those files are included
+         as Resource, Content then remove them from the Page items.
+         
+         If EnableDefaultApplicationDefinition is false, ApplicationDefinition must also be removed from Page as
+         PresentationBuildTasks will include the generated code for the ApplicationDefinition twice otherwise.
+         We cannot do this in all cases in the props file as ApplicationDefinition won't be available
+         at that point in the evaluation when EnableDefaultApplicationDefinition is false.
+    -->
+    <Page Remove="@(Resource);@(Content);@(ApplicationDefinition)"
```

Comment:
We should remove `@(ApplicationDefinition)` unconditionally - otherwise project-supplied value for `@(ApplicationDefinition)` will not be removed here. I would propose something like this: 

 ```xml

<Page Remove="@(Resource);@(Content)" Condition="'$(EnableDefaultPageItems)'!='false'" />
<Page Remove="@(ApplicationDefinition)" />
```

Reply:
Forgot to make that change this morning.